src/soroban.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Soroban helper class
+ * formatting, parsing, and etc
+ * @class Soroban
+ */
+export class Soroban {
+  /**
+   * All arithmetic inside the contract is performed on integers to
+   * avoid potential precision and consistency issues of floating-point
+   *
+   * This function takes the smart contract value and its decimals (if the token has any) and returns a display value
+   * @param {string} amount - the token amount you want to display
+   * @param {number} decimals - specify how many decimal places a token has
+   * @returns {string} - display value
+   */
+  static formatTokenAmount(amount, decimals) {
+    let formatted = amount;
+
+    if (amount.includes('.')) {
+      throw new Error('No decimal is allowed');
+    }
+
+    if (decimals > 0) {
+      if (decimals > formatted.length) {
+        const targetShiftedLength = decimals - formatted.length;
+        const targetTotalLength = targetShiftedLength + formatted.length;
+        formatted = [
+          '0',
+          formatted.toString().padStart(targetTotalLength, '0')
```

Comment:
what ya think @Shaptic

Reply:
hmm... I'm expanding this math and it sounds like: 

```js
targetTotalLength = (decimals - formatted.length) + formatted.length;
```

so that's just `decimals`? Am I missing something?

Reply:
lol... :homerhide:

Reply:
fixed